{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "import lxml.etree\n",
    "from nltk.tokenize import sent_tokenize\n",
    "import re\n",
    "from nltk.parse.stanford import StanfordDependencyParser\n",
    "path_to_jar = 'resources/parser/stanford-corenlp-3.9.1.jar'\n",
    "path_to_models_jar = 'resources/parser/stanford-corenlp-3.9.1-models.jar'\n",
    "dependency_parser = StanfordDependencyParser(path_to_jar=path_to_jar, path_to_models_jar=path_to_models_jar)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "with open('datasets/Biomed/aimed_bioc.xml', 'r') as myfile:\n",
    "    data=myfile.read().replace('\\n', '').replace('.', '. ')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "root = lxml.etree.fromstring(data)\n",
    "documents = root.findall('document')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def protein_extraction(document):\n",
    "    annotations = document.findall('passage/annotation')\n",
    "    protein_dict = {}\n",
    "    for annotation in annotations:\n",
    "        protein_dict[annotation.attrib['id']] = annotation.find('text').text\n",
    "    return protein_dict"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def relation_extraction(document):\n",
    "    relations = document.findall('passage/relation')\n",
    "    relation_pairs = []\n",
    "    for relation in relations:\n",
    "        nodes = relation.findall('node')\n",
    "        relation_pairs.append((nodes[0].attrib['refid'], nodes[1].attrib['refid']))\n",
    "    return relation_pairs      "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def extract_root(sentence):\n",
    "    result = dependency_parser.raw_parse(sentence)\n",
    "    dep = result.__next__()\n",
    "    return list(dep.triples())[0][0][0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Isolation of human delta-catenin and its binding specificity with presenilin 1.\n",
      "Isolation\n",
      "We screened proteins for interaction with presenilin (PS) 1, and cloned the full-length cDNA of human delta-catenin, which encoded 1225 amino acids.\n",
      "screened\n",
      "Yeast two-hybrid assay, GST binding assay and immunoprecipitation demonstrated that delta-catenin interacted with a hydrophilic loop region in the endoproteolytic C-terminal fragment of PS1, but not with that of PS-2.\n",
      "demonstrated\n",
      "We also found another armadillo-protein, p0071, interacted with PS1.\n",
      "found\n",
      "-----\n",
      "Activation of interferon-gamma inducing factor mediated by interleukin-1beta converting enzyme.\n",
      "mediated\n",
      "The interleukin-1beta (IL-1beta) converting enzyme (ICE) processes the inactive IL-1beta precursor to the proinflammatory cytokine.\n",
      "processes\n",
      "The interleukin-1beta (IL-1beta) converting enzyme (ICE) processes the inactive IL-1beta precursor to the proinflammatory cytokine.\n",
      "processes\n",
      "ICE was also shown to cleave the precursor of interferon-gamma inducing factor (IGIF) at the authentic processing site with high efficiency, thereby activating IGIF and facilitating its export.\n",
      "shown\n",
      "ICE was also shown to cleave the precursor of interferon-gamma inducing factor (IGIF) at the authentic processing site with high efficiency, thereby activating IGIF and facilitating its export.\n",
      "shown\n",
      "ICE was also shown to cleave the precursor of interferon-gamma inducing factor (IGIF) at the authentic processing site with high efficiency, thereby activating IGIF and facilitating its export.\n",
      "shown\n",
      "Lipopolysaccharide-activated ICE-deficient (ICE-/-) Kupffer cells synthesized the IGIF precursor but failed to process it into the active form.\n",
      "synthesized\n",
      "Lipopolysaccharide-activated ICE-deficient (ICE-/-) Kupffer cells synthesized the IGIF precursor but failed to process it into the active form.\n",
      "synthesized\n",
      "Interferon-gamma and IGIF were diminished in the sera of ICE-/- mice exposed to Propionibacterium acnes and lipopolysaccharide.\n",
      "diminished\n",
      "Interferon-gamma and IGIF were diminished in the sera of ICE-/- mice exposed to Propionibacterium acnes and lipopolysaccharide.\n",
      "diminished\n",
      "-----\n",
      "The results showed that IL-11 and IL-6 can both stimulate cell proliferation, induce similar pattern of protein tyrosine phosphorylation, and activate the same proto-oncogene (junB) expression in TF-1 cells.\n",
      "showed\n",
      "The results showed that IL-11 and IL-6 can both stimulate cell proliferation, induce similar pattern of protein tyrosine phosphorylation, and activate the same proto-oncogene (junB) expression in TF-1 cells.\n",
      "showed\n",
      "It was observed that anti-gp130 antibodies abolished cell proliferation, protein tyrosine phosphorylation, and junB gene expression elicited by IL-11 or IL-6 in TF-1 cells.\n",
      "observed\n",
      "It was observed that anti-gp130 antibodies abolished cell proliferation, protein tyrosine phosphorylation, and junB gene expression elicited by IL-11 or IL-6 in TF-1 cells.\n",
      "observed\n",
      "-----\n",
      "Interaction of mammalian splicing factor SF3a with U2 snRNP and relation of its 60-kD subunit to yeast PRP9.\n",
      "Interaction\n",
      "Interaction of mammalian splicing factor SF3a with U2 snRNP and relation of its 60-kD subunit to yeast PRP9.\n",
      "Interaction\n",
      "Here it is shown that the SF3a subunits correspond to three of the 17S U2 snRNP-specific polypeptides.\n",
      "shown\n",
      "Here it is shown that the SF3a subunits correspond to three of the 17S U2 snRNP-specific polypeptides.\n",
      "shown\n",
      "SF3a interacts with U2 snRNP in the presence of SF3b to generate a structure similar to 17S U2 snRNP, which suggests a function for SF3a and SF3b in the incorporation of U2 snRNP into the spliceosome.\n",
      "interacts\n",
      "SF3a interacts with U2 snRNP in the presence of SF3b to generate a structure similar to 17S U2 snRNP, which suggests a function for SF3a and SF3b in the incorporation of U2 snRNP into the spliceosome.\n",
      "interacts\n",
      "-----\n",
      "CCR5 binds multiple CC-chemokines: MCP-3 acts as a natural antagonist.\n",
      "binds\n",
      "CCR5 binds multiple CC-chemokines: MCP-3 acts as a natural antagonist.\n",
      "binds\n",
      "CCR5 was first characterized as a receptor for MIP-1alpha, MIP-1beta, and RANTES, and was rapidly shown to be the main coreceptor for M-tropic human immunodeficiency virus (HIV)-1 strains and simian immunodeficiency virus (SIV).\n",
      "characterized\n",
      "CCR5 was first characterized as a receptor for MIP-1alpha, MIP-1beta, and RANTES, and was rapidly shown to be the main coreceptor for M-tropic human immunodeficiency virus (HIV)-1 strains and simian immunodeficiency virus (SIV).\n",
      "characterized\n",
      "CCR5 was first characterized as a receptor for MIP-1alpha, MIP-1beta, and RANTES, and was rapidly shown to be the main coreceptor for M-tropic human immunodeficiency virus (HIV)-1 strains and simian immunodeficiency virus (SIV).\n",
      "characterized\n",
      "CCR5 was first characterized as a receptor for MIP-1alpha, MIP-1beta, and RANTES, and was rapidly shown to be the main coreceptor for M-tropic human immunodeficiency virus (HIV)-1 strains and simian immunodeficiency virus (SIV).\n",
      "characterized\n",
      "CCR5 was first characterized as a receptor for MIP-1alpha, MIP-1beta, and RANTES, and was rapidly shown to be the main coreceptor for M-tropic human immunodeficiency virus (HIV)-1 strains and simian immunodeficiency virus (SIV).\n",
      "characterized\n",
      "CCR5 was first characterized as a receptor for MIP-1alpha, MIP-1beta, and RANTES, and was rapidly shown to be the main coreceptor for M-tropic human immunodeficiency virus (HIV)-1 strains and simian immunodeficiency virus (SIV).\n",
      "characterized\n",
      "CCR5 was first characterized as a receptor for MIP-1alpha, MIP-1beta, and RANTES, and was rapidly shown to be the main coreceptor for M-tropic human immunodeficiency virus (HIV)-1 strains and simian immunodeficiency virus (SIV).\n",
      "characterized\n",
      "All CC-chemokines currently available were tested for their ability to compete with [(125)I]-MIP-1beta binding on a stable cell line expressing recombinant CCR5, and/or to induce a functional response in these cells.\n",
      "tested\n",
      "All CC-chemokines currently available were tested for their ability to compete with [(125)I]-MIP-1beta binding on a stable cell line expressing recombinant CCR5, and/or to induce a functional response in these cells.\n",
      "tested\n",
      "All CC-chemokines currently available were tested for their ability to compete with [(125)I]-MIP-1beta binding on a stable cell line expressing recombinant CCR5, and/or to induce a functional response in these cells.\n",
      "tested\n",
      "We found that in addition to MIP-1beta, MIP-1alpha, and RANTES, five other CC-chemokines could compete for [(125)I]-MIP-1beta binding: MCP-2, MCP-3, MCP-4, MCP-1, and eotaxin binding was characterized by IC(50) values of 0.\n",
      "characterized\n",
      "We found that in addition to MIP-1beta, MIP-1alpha, and RANTES, five other CC-chemokines could compete for [(125)I]-MIP-1beta binding: MCP-2, MCP-3, MCP-4, MCP-1, and eotaxin binding was characterized by IC(50) values of 0.\n",
      "characterized\n",
      "We found that in addition to MIP-1beta, MIP-1alpha, and RANTES, five other CC-chemokines could compete for [(125)I]-MIP-1beta binding: MCP-2, MCP-3, MCP-4, MCP-1, and eotaxin binding was characterized by IC(50) values of 0.\n",
      "characterized\n",
      "We found that in addition to MIP-1beta, MIP-1alpha, and RANTES, five other CC-chemokines could compete for [(125)I]-MIP-1beta binding: MCP-2, MCP-3, MCP-4, MCP-1, and eotaxin binding was characterized by IC(50) values of 0.\n",
      "characterized\n",
      "We found that in addition to MIP-1beta, MIP-1alpha, and RANTES, five other CC-chemokines could compete for [(125)I]-MIP-1beta binding: MCP-2, MCP-3, MCP-4, MCP-1, and eotaxin binding was characterized by IC(50) values of 0.\n",
      "characterized\n",
      "We found that in addition to MIP-1beta, MIP-1alpha, and RANTES, five other CC-chemokines could compete for [(125)I]-MIP-1beta binding: MCP-2, MCP-3, MCP-4, MCP-1, and eotaxin binding was characterized by IC(50) values of 0.\n",
      "characterized\n",
      "We found that in addition to MIP-1beta, MIP-1alpha, and RANTES, five other CC-chemokines could compete for [(125)I]-MIP-1beta binding: MCP-2, MCP-3, MCP-4, MCP-1, and eotaxin binding was characterized by IC(50) values of 0.\n",
      "characterized\n",
      "We found that in addition to MIP-1beta, MIP-1alpha, and RANTES, five other CC-chemokines could compete for [(125)I]-MIP-1beta binding: MCP-2, MCP-3, MCP-4, MCP-1, and eotaxin binding was characterized by IC(50) values of 0.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "characterized\n",
      "Among these ligands, MCP-3 had the remarkable property of binding CCR5 with high affinity without eliciting a functional response, MCP-3 could also inhibit the activation of CCR5 by MIP-1beta and may therefore be considered as a natural antagonist for CCR5.\n",
      "inhibit\n",
      "Among these ligands, MCP-3 had the remarkable property of binding CCR5 with high affinity without eliciting a functional response, MCP-3 could also inhibit the activation of CCR5 by MIP-1beta and may therefore be considered as a natural antagonist for CCR5.\n",
      "inhibit\n",
      "Among these ligands, MCP-3 had the remarkable property of binding CCR5 with high affinity without eliciting a functional response, MCP-3 could also inhibit the activation of CCR5 by MIP-1beta and may therefore be considered as a natural antagonist for CCR5.\n",
      "inhibit\n",
      "Among these ligands, MCP-3 had the remarkable property of binding CCR5 with high affinity without eliciting a functional response, MCP-3 could also inhibit the activation of CCR5 by MIP-1beta and may therefore be considered as a natural antagonist for CCR5.\n",
      "inhibit\n",
      "Among these ligands, MCP-3 had the remarkable property of binding CCR5 with high affinity without eliciting a functional response, MCP-3 could also inhibit the activation of CCR5 by MIP-1beta and may therefore be considered as a natural antagonist for CCR5.\n",
      "inhibit\n",
      "Among these ligands, MCP-3 had the remarkable property of binding CCR5 with high affinity without eliciting a functional response, MCP-3 could also inhibit the activation of CCR5 by MIP-1beta and may therefore be considered as a natural antagonist for CCR5.\n",
      "inhibit\n",
      "Chemokines that could compete with high affinity for MIP-1beta binding could also compete for monomeric gp120 binding, although with variable potencies; maximal gp120 binding inhibition was 80% for MCP-2, but only 30% for MIP-1beta.\n",
      "compete\n",
      "Chemokines that could compete with high affinity for MIP-1beta binding could also compete for monomeric gp120 binding, although with variable potencies; maximal gp120 binding inhibition was 80% for MCP-2, but only 30% for MIP-1beta.\n",
      "compete\n",
      "Chemokines that could compete with high affinity for MIP-1beta binding could also compete for monomeric gp120 binding, although with variable potencies; maximal gp120 binding inhibition was 80% for MCP-2, but only 30% for MIP-1beta.\n",
      "compete\n",
      "Chemokines that could compete with high affinity for MIP-1beta binding could also compete for monomeric gp120 binding, although with variable potencies; maximal gp120 binding inhibition was 80% for MCP-2, but only 30% for MIP-1beta.\n",
      "compete\n",
      "MCP-3 could compete efficiently for gp120 binding but was, however, found to be a weak inhibitor of HIV infection, probably as a consequence of its inability to downregulate the receptor.\n",
      "compete\n",
      "-----\n",
      "Competitive binding of alpha-actinin and calmodulin to the NMDA receptor.\n",
      "binding\n",
      "Competitive binding of alpha-actinin and calmodulin to the NMDA receptor.\n",
      "binding\n",
      "alpha-Actinin-2, a member of the spectrin/dystrophin family of actin-binding proteins, is identified here as a brain postsynaptic density protein that colocalizes in dendritic spines with NMDA receptors and the putative NMDA receptor-clustering molecule PSD-95.\n",
      "identified\n",
      "alpha-Actinin-2, a member of the spectrin/dystrophin family of actin-binding proteins, is identified here as a brain postsynaptic density protein that colocalizes in dendritic spines with NMDA receptors and the putative NMDA receptor-clustering molecule PSD-95.\n",
      "identified\n",
      "alpha-Actinin-2 binds by its central rod domain to the cytoplasmic tail of both NR1 and NR2B subunits of the NMDA receptor, and can be immunoprecipitated with NMDA receptors and PSD-95 from rat brain.\n",
      "binds\n",
      "alpha-Actinin-2 binds by its central rod domain to the cytoplasmic tail of both NR1 and NR2B subunits of the NMDA receptor, and can be immunoprecipitated with NMDA receptors and PSD-95 from rat brain.\n",
      "binds\n",
      "Thus alpha-actinin may play a role in both the localization of NMDA receptors and their modulation by Ca2+.\n",
      "play\n",
      "-----\n",
      "The retinoblastoma protein associates with the protein phosphatase type 1 catalytic subunit.\n",
      "phosphatase\n",
      "In vitro-binding assays demonstrated that PP-1 alpha isoforms preferentially bind the hypophosphorylated form of p110RB.\n",
      "demonstrated\n",
      "Moreover, similar p110RB sequences are required for binding PP-1 alpha 2 and SV40 large T antigen.\n",
      "required\n",
      "Moreover, similar p110RB sequences are required for binding PP-1 alpha 2 and SV40 large T antigen.\n",
      "required\n",
      "Moreover, similar p110RB sequences are required for binding PP-1 alpha 2 and SV40 large T antigen.\n",
      "required\n",
      "-----\n",
      "A region of the beta subunit of the interferon alpha receptor different from box 1 interacts with Jak1 and is sufficient to activate the Jak-Stat pathway and induce an antiviral state.\n",
      "interacts\n",
      "-----\n",
      "A mutant form of human interferon-gamma (IFN-gamma SC1) that binds one IFN-gamma receptor alpha chain (IFN-gamma R alpha) has been designed and characterized.\n",
      "chain\n",
      "A mutant form of human interferon-gamma (IFN-gamma SC1) that binds one IFN-gamma receptor alpha chain (IFN-gamma R alpha) has been designed and characterized.\n",
      "chain\n",
      "A mutant form of human interferon-gamma (IFN-gamma SC1) that binds one IFN-gamma receptor alpha chain (IFN-gamma R alpha) has been designed and characterized.\n",
      "chain\n",
      "A mutant form of human interferon-gamma (IFN-gamma SC1) that binds one IFN-gamma receptor alpha chain (IFN-gamma R alpha) has been designed and characterized.\n",
      "chain\n",
      "A mutant form of human interferon-gamma (IFN-gamma SC1) that binds one IFN-gamma receptor alpha chain (IFN-gamma R alpha) has been designed and characterized.\n",
      "chain\n",
      "A mutant form of human interferon-gamma (IFN-gamma SC1) that binds one IFN-gamma receptor alpha chain (IFN-gamma R alpha) has been designed and characterized.\n",
      "chain\n",
      "Isothermal titration calorimetry shows that IFN-gamma SC1 forms a 1:1 complex with its high-affinity receptor (IFN-gamma R alpha) with an affinity of 27(+/- 9) nM.\n",
      "shows\n",
      "Isothermal titration calorimetry shows that IFN-gamma SC1 forms a 1:1 complex with its high-affinity receptor (IFN-gamma R alpha) with an affinity of 27(+/- 9) nM.\n",
      "shows\n",
      "Isothermal titration calorimetry shows that IFN-gamma SC1 forms a 1:1 complex with its high-affinity receptor (IFN-gamma R alpha) with an affinity of 27(+/- 9) nM.\n",
      "shows\n",
      "Isothermal titration calorimetry shows that IFN-gamma SC1 forms a 1:1 complex with its high-affinity receptor (IFN-gamma R alpha) with an affinity of 27(+/- 9) nM.\n",
      "shows\n",
      "Isothermal titration calorimetry shows that IFN-gamma SC1 forms a 1:1 complex with its high-affinity receptor (IFN-gamma R alpha) with an affinity of 27(+/- 9) nM.\n",
      "shows\n",
      "As a result, surface charge rather than structural changes is likely responsible for the inability of the His111-->Asp domain of to bind IFN-gamma R alpha.\n",
      "responsible\n",
      "As a result, surface charge rather than structural changes is likely responsible for the inability of the His111-->Asp domain of to bind IFN-gamma R alpha.\n",
      "responsible\n",
      "The AB loops of IFN-gamma SC1 adopt conformations similar to the ordered loops of IFN-gamma observed in the crystal structure of the IFN-gamma/IFN-gamma R alpha complex.\n",
      "adopt\n",
      "The AB loops of IFN-gamma SC1 adopt conformations similar to the ordered loops of IFN-gamma observed in the crystal structure of the IFN-gamma/IFN-gamma R alpha complex.\n",
      "adopt\n",
      "The AB loops of IFN-gamma SC1 adopt conformations similar to the ordered loops of IFN-gamma observed in the crystal structure of the IFN-gamma/IFN-gamma R alpha complex.\n",
      "adopt\n",
      "The AB loops of IFN-gamma SC1 adopt conformations similar to the ordered loops of IFN-gamma observed in the crystal structure of the IFN-gamma/IFN-gamma R alpha complex.\n",
      "adopt\n",
      "The AB loops of IFN-gamma SC1 adopt conformations similar to the ordered loops of IFN-gamma observed in the crystal structure of the IFN-gamma/IFN-gamma R alpha complex.\n",
      "adopt\n",
      "Thus, IFN-gamma R alpha binding does not result in a large conformational change in the AB loop as previously suggested.\n",
      "result\n",
      "Thus, IFN-gamma R alpha binding does not result in a large conformational change in the AB loop as previously suggested.\n",
      "result\n",
      "Despite binding to only one IFN-gamma R alpha, IFN-gamma SC1 is biologically active in cell proliferation, MHC class I induction, and anti-viral assays.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "active\n",
      "Despite binding to only one IFN-gamma R alpha, IFN-gamma SC1 is biologically active in cell proliferation, MHC class I induction, and anti-viral assays.\n"
     ]
    },
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-53-ea9d12e47a0c>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m     13\u001b[0m     \u001b[0;32mfor\u001b[0m \u001b[0msentence\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mfiltered_sentences\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     14\u001b[0m         \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msentence\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 15\u001b[0;31m         \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mextract_root\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msentence\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     16\u001b[0m     \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'-----'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m<ipython-input-46-f5942408e44b>\u001b[0m in \u001b[0;36mextract_root\u001b[0;34m(sentence)\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[0;32mdef\u001b[0m \u001b[0mextract_root\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msentence\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 2\u001b[0;31m     \u001b[0mresult\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mdependency_parser\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mraw_parse\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msentence\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      3\u001b[0m     \u001b[0mdep\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mresult\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m__next__\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      4\u001b[0m     \u001b[0;32mreturn\u001b[0m \u001b[0mlist\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdep\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mtriples\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.6/site-packages/nltk/parse/stanford.py\u001b[0m in \u001b[0;36mraw_parse\u001b[0;34m(self, sentence, verbose)\u001b[0m\n\u001b[1;32m    132\u001b[0m         \u001b[0;34m:\u001b[0m\u001b[0mrtype\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0miter\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mTree\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    133\u001b[0m         \"\"\"\n\u001b[0;32m--> 134\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0mnext\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mraw_parse_sents\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0msentence\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mverbose\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    135\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    136\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mraw_parse_sents\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0msentences\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mverbose\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mFalse\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.6/site-packages/nltk/parse/stanford.py\u001b[0m in \u001b[0;36mraw_parse_sents\u001b[0;34m(self, sentences, verbose)\u001b[0m\n\u001b[1;32m    150\u001b[0m             \u001b[0;34m'-outputFormat'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_OUTPUT_FORMAT\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    151\u001b[0m         ]\n\u001b[0;32m--> 152\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_parse_trees_output\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_execute\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcmd\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'\\n'\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mjoin\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msentences\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mverbose\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    153\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    154\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mtagged_parse\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0msentence\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mverbose\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mFalse\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.6/site-packages/nltk/parse/stanford.py\u001b[0m in \u001b[0;36m_execute\u001b[0;34m(self, cmd, input_, verbose)\u001b[0m\n\u001b[1;32m    216\u001b[0m                 \u001b[0mcmd\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mappend\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0minput_file\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mname\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    217\u001b[0m                 stdout, stderr = java(cmd, classpath=self._classpath,\n\u001b[0;32m--> 218\u001b[0;31m                                       stdout=PIPE, stderr=PIPE)\n\u001b[0m\u001b[1;32m    219\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    220\u001b[0m             \u001b[0mstdout\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mstdout\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mreplace\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34mb'\\xc2\\xa0'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34mb' '\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.6/site-packages/nltk/__init__.py\u001b[0m in \u001b[0;36mjava\u001b[0;34m(cmd, classpath, stdin, stdout, stderr, blocking)\u001b[0m\n\u001b[1;32m    129\u001b[0m     \u001b[0mp\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0msubprocess\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mPopen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcmd\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mstdin\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mstdin\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mstdout\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mstdout\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mstderr\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mstderr\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    130\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0mblocking\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0;32mreturn\u001b[0m \u001b[0mp\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 131\u001b[0;31m     \u001b[0;34m(\u001b[0m\u001b[0mstdout\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mstderr\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcommunicate\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    132\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    133\u001b[0m     \u001b[0;31m# Check the return code.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.6/subprocess.py\u001b[0m in \u001b[0;36mcommunicate\u001b[0;34m(self, input, timeout)\u001b[0m\n\u001b[1;32m    841\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    842\u001b[0m             \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 843\u001b[0;31m                 \u001b[0mstdout\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mstderr\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_communicate\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0minput\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mendtime\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mtimeout\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    844\u001b[0m             \u001b[0;32mfinally\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    845\u001b[0m                 \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_communication_started\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mTrue\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.6/subprocess.py\u001b[0m in \u001b[0;36m_communicate\u001b[0;34m(self, input, endtime, orig_timeout)\u001b[0m\n\u001b[1;32m   1512\u001b[0m                         \u001b[0;32mraise\u001b[0m \u001b[0mTimeoutExpired\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0morig_timeout\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1513\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1514\u001b[0;31m                     \u001b[0mready\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mselector\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mselect\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mtimeout\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1515\u001b[0m                     \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_check_timeout\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mendtime\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0morig_timeout\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1516\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.6/selectors.py\u001b[0m in \u001b[0;36mselect\u001b[0;34m(self, timeout)\u001b[0m\n\u001b[1;32m    374\u001b[0m             \u001b[0mready\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    375\u001b[0m             \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 376\u001b[0;31m                 \u001b[0mfd_event_list\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_poll\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mpoll\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mtimeout\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    377\u001b[0m             \u001b[0;32mexcept\u001b[0m \u001b[0mInterruptedError\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    378\u001b[0m                 \u001b[0;32mreturn\u001b[0m \u001b[0mready\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "for document in documents:\n",
    "    proteins = protein_extraction(document)\n",
    "    relations = relation_extraction(document)\n",
    "    sentences = sent_tokenize(document.find('passage/text').text)\n",
    "    \n",
    "    filtered_sentences = []\n",
    "    for sentence in sentences:\n",
    "        for relation in relations:\n",
    "            if proteins[relation[0]] in sentence and proteins[relation[1]] in sentence:\n",
    "                filtered_sentences.append(sentence)\n",
    "                \n",
    "                \n",
    "    for sentence in filtered_sentences:\n",
    "        print(sentence)\n",
    "        print(extract_root(sentence))\n",
    "    print('-----')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
